sync.py

```diff
@@ -99,6 +103,40 @@ def check_unsynced_modules(tagged_builds, modules_to_track):
 
     return unsynced_builds
 
+def check_cve_builds(tagged_builds):
+    """
+    Look for builds that aren't allowed and filter them
+    """
+    if not filter_cve:
+        return tagged_builds
+    import access
+
+    reqs = {}
+    for build in sorted(tagged_builds, key=lambda x: x['package_name']):
+        if build['nvr'] in reqs:
+            continue
+
+        n, v, r = build['nvr'].rsplit('-', 2)
+        req = access.NvrInfo(n, v, r)
+        reqs[build['nvr']] = req
+        #  Precache for speed, downside is it means once we get an allow
+        # we stop querying.
+        if not req.hist_precache():
```

Comment:
We plan to run this script through a cronjob, so we should hit each package only once. In what case this sort of caching would be useful ?

Reply:
The way the CVE checker works is that you do a request which kicks off a job that checks bugzilla to see if there are any CVEs open against the package, the return from the request is an ID and you then repeatedly check this ID to see when the request has finished (I'm not sure, but I think this scales with the number of open bugs against the package). This can take a _long_ time, current timeout is 20 minutes per. NVR ... but that might not be long enough in some cases (Eg. kernel).

This does a different call to the CVE checker to see if we've already got a result for the NVR, and if so it uses that instead of the above ... success or fail, this takes a couple of seconds.

Reply:
It seems the cve checker service is very slow. We should talk the people behind this service and tell them that we only care about checking bugs that have not been shipped in CentOS already. They should only check for diff of bugs that have not been shipped to CentOS (instead of checking for all the bugs that are in changelog of a package).